internal/wsecho/wsecho.go

```diff
@@ -0,0 +1,73 @@
+// +build !js
+
+package wsecho
+
+import (
+	"context"
+	"io"
+	"log"
+	"net/http"
+	"time"
+
+	"nhooyr.io/websocket"
+)
+
+// Serve provides a streaming WebSocket echo server
+// for use in tests.
+func Serve(w http.ResponseWriter, r *http.Request) {
+	c, err := websocket.Accept(w, r, &websocket.AcceptOptions{
+		Subprotocols:       []string{"echo"},
+		InsecureSkipVerify: true,
+	})
+	if err != nil {
+		log.Printf("echo server: failed to accept: %+v", err)
+		return
+	}
+	defer c.Close(websocket.StatusInternalError, "")
+
+	Loop(r.Context(), c)
+}
+
+// Loop echos every msg received from c until an error
+// occurs or the context expires.
+// The read limit is set to 1 << 40.
+func Loop(ctx context.Context, c *websocket.Conn) {
+	defer c.Close(websocket.StatusInternalError, "")
+
+	c.SetReadLimit(1 << 40)
+
+	ctx, cancel := context.WithTimeout(ctx, time.Minute)
+	defer cancel()
+
+	b := make([]byte, 32768)
```

Comment:
Since you used `1 << 40` above `32 << 10` would be consistent here

Reply:
Also just realized 1 << 40 is a terabyte, not a gigabyte lol.